Dockerfile

```diff
@@ -43,13 +14,13 @@ WORKDIR /multivac
 
 # set up bdist_wheel
 RUN pip3 install wheel
-
 RUN pip3 install setuptools
-RUN pip3 install torch==1.2.0
 
-# env setup for tensorflow
+# env setup for torch + other requirements
+RUN pip3 install torch==1.2.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
 RUN pip3 install -r requirements.txt
```

Comment:
Guys - this won't work for us, as we need `torch` with GPU capabilities. This specifies that it's only downloading the CPU version. Is that `cpu` flag required for this to work?

Reply:
@gallupBenRyan  -- 

Yes, the `cpu` flag is required for this to work in its current state.  We are bypassing the necessity of `CUDA`; please see here:

https://pytorch.org/get-started/locally/

Also, I think we are confusing `pytorch`, `torchvision`, and `torch`. What exactly are all of those? 

If we need `CUDA` alongside `GPU` requirements (which are currently not working, I thought?) we would need to revamp the Python environment here to Conda.  I did not know that up front.

Reply:
CUDA/GPU is working, it's just not providing a significant performance boost over CPU because the tree-based nature of the models is not very parallelizable. One of the more immediate next steps for us is to implement some more parallelization to overcome that, but regardless - yes, GPU capabilities are important to include here.

For clarity, `pytorch` is the name of the package when installing via Anaconda and `torch` is for installing via pip (i.e. `pip3 install torch==1.2.0`). `torchvision` is a package of datasets and functions for working with images/videos for computer vision - not necessary for us here. 

Anaconda is not required for CUDA/GPU processing, so we don't need to revamp anything for that reason. 

Can we try these pip commands with the `--no-cache-dir` flag? I'm wondering if that would help with the memory/space issues.

Reply:
Saving this to reference later:

https://stackoverflow.com/questions/57121936/changing-a-dockerfile-from-cuda9-to-cuda10